[KEYCLOAK-17315] [KEYCLOAK-17326] Fixes

iankko opened the pull request:
<hr/>

    [KEYCLOAK-17315] Enable run of clustering and Cross-DC tests on Wildfly
    within GitHub actions also for changes upgrading Keycloak to next
    Wildfly version
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

_Note:_ Would be helpful to get this one reviewed / merged yet before the https://github.com/keycloak/keycloak/pull/7705 one, so it's not necessary to manually re-run clustering / Cross-DC tests on each rebase of the `Upgrade to Wildlfy 22` one.

<hr/>

    [KEYCLOAK-17326] Fix:
    
    $ git diff --name-only HEAD^
    fatal: ambiguous argument 'HEAD^': unknown revision or path not in the working tree.
    Use '--' to separate paths from revisions, like this:
    'git <command> [<revision>...] -- [<file>...]'
    
    GHA failure on 'Test Clustering on Wildfly' phase. See e.g. recent:
      https://github.com/keycloak/keycloak/pull/7705/checks?check_run_id=2023996258
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

<hr/>

<!---
Please read https://github.com/keycloak/keycloak/blob/master/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

Commit:
[KEYCLOAK-17315] Enable run of clustering and Cross-DC tests on Wildfly
within GitHub actions also for changes upgrading Keycloak to next
Wildfly version

Also, update GIT_DIFF evaluation per Hynek's suggestion. Thanks, Hynek!

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>
Co-authored-by: Hynek Mlnařík <hmlnarik@users.noreply.github.com>

iankko commented:
@pskopek @hmlnarik @mposolda @pedroigor @stianst PTAL once got a chance

Thanks, Jan

hmlnarik commented on .github/workflows/ci.yml:
```diff
@@ -187,9 +187,11 @@ jobs:
 
       - name: Check whether this phase should run
-        run: echo "GIT_DIFF=$( git diff --name-only HEAD^ | egrep -ic 'crossdc|infinispan' )" >> $GITHUB_ENV
+        run: echo "GIT_DIFF=$(( $( git diff --name-only HEAD^ | egrep -ic 'crossdc|infinispan' ) | $( git diff HEAD^ pom.xml | egrep -ic '\+\s+<wildfly.version>' ) ))" >> $GITHUB_ENV
```
Did you mean the following?
```suggestion
        run: echo "GIT_DIFF=$[ $( git diff --name-only HEAD^ | egrep -ic 'crossdc|infinispan' ) + $( git diff HEAD^ pom.xml | egrep -ic '\+\s+<wildfly.version>' ) ]" >> $GITHUB_ENV
```

iankko replied:
Yeah, that would work too to get `GIT_DIFF` to evaluate to non-zero if Wildfly version got bumped. And is actually more readable due not needing to add that much brackets.

@hmlnarik Thanks for the suggestion! Added it into the 1-th commit & squashed.

Commit:
[KEYCLOAK-17326] Fix:

$ git diff --name-only HEAD^
fatal: ambiguous argument 'HEAD^': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

GHA failure on 'Test Clustering on Wildfly' phase. See e.g. recent:
  https://github.com/keycloak/keycloak/pull/7705/checks?check_run_id=2023996258

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

hmlnarik reviewed (approved):
Thanks @iankko!
